### PR TITLE
Makes the recycler slightly less fuck because it is 68% too much fuck.

### PR DESCRIPTION
**Pull request by Buggy123**
Remake of #48299 because Oranges ran out of juice before finishing it.

See issue #48706 for level of fuck. It's not great and one of them is a trivial server-crashing exploit. I couldn't fix all of it without reworking how it worked at least slightly, because it was so broken that half of the features were bugs and half of the features weren't there because bugs.

Summary of the changed behavior.

Still recycles and outputs the materials, etc etc. That is unchanged except for consistency.

Containers with lots of contents will no longer break everything when recycled.

MMIs, posibrains, plain-old organic brains, and mobs will cause a emergency stop when they enter the recycler, and if inside a backpack or bodybag or such the backpack, living thing, and all other contents will remain intact and unchanged.

In certain scenarios, if the level of fuck was low enough for it to work at all, the recycler would play one sound per item that entered it (including box contents), easily resulting in massive spam. This has been greatly reduced and to compensate it just gets louder if it recycles more things at once.

If emagged, MMIs, posibrains, and brain-brains will be consumed and destroyed completely. Mobs will behave as before; all worn/held items are recycled and the mob is dealt 1,000 brute damage, which usually kills it. If a mob is inside of any sort of container (bluespace body bag), the container is recycled and the mob is crushed and left on the ground as normal. Notably, the var to make a emagged recycler _not_ eat a victim's items has been removed. It was unused and retaining it would've been a pain.

🆑 
fix: After a 96 hour brainstorming session involving eighty interns, half of the engineering department, three contractors from the Wizard Federation, and a priest, the recycler has been re-jonglulated. Operators should find it to operate more consistently, efficiently, safely, loudly, and with almost absolutely no snaring of the local space-time continuum.
/ 🆑 

I've been working on this for four days straight. It's tested, it works efficiently even with very large amounts of items, it doesn't crash the server anymore, please merge.

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
One gaussian standard deviation of fuck eh? Impressive :1st_place_medal:

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Fixup the conflicts. Also why are we moving things to nullspace before deleting them?

**Comment by Buggy123** [comment]
>Fixup the conflicts.

On it.

>Also why are we moving things to nullspace before deleting them?

Maintainer recommendation:
image

The second best way was basically copying GetAllContents(), because while it works well in its current state because it outputs a ordered list, the order isn't part of the spec so it might change in the future.

**Comment by Buggy123** [comment]
Also yeah merge conflict is resolved.

**Comment by Buggy123** [comment]
Shush, it's not stale it's waiting for approval and merging.

**Comment by Tharcoonvagh** [comment]
s t a l e gang

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -90,66 +90,80 @@
...
+	for(var/nommed in nom)
+		recycle_item(nommed)
+	if(nom.len && sound)
+		playsound(src, item_recycle_sound, (50 + nom.len*5), TRUE, nom.len, ignore_walls = (nom.len - 10)) // As a substitute for playing 50 sounds at once.
+	if(not_eaten)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', (50 + not_eaten*5), FALSE, not_eaten, ignore_walls = (not_eaten - 10)) // Ditto.
+	if(!isliving(AM0))
+		AM0.moveToNullspace()
+		qdel(AM0)
+	else // Lets not move a mob to nullspace and qdel it, yes?
```
Comment (Cyberboss) [inline code/game/machinery/recycler.dm]: Why not ismob then?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/recycler.dm (hunk 1 above)
Comment (Buggy123) [reply code/game/machinery/recycler.dm]: because i'm stupid

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**